howtos/otel/how_to_integrate_opentelemetry_with_pubsubplus_api.py

```diff
@@ -0,0 +1,150 @@
+from solace.messaging.resources.topic_subscription import TopicSubscription
+from solace.messaging.resources.queue import Queue
+from solace.messaging.receiver.message_receiver import MessageHandler
+from solace.messaging.resources.topic import Topic
+from solace.messaging.receiver.inbound_message import InboundMessage
+from solace.messaging.messaging_service import MessagingService
+from solace.messaging.receiver.persistent_message_receiver import PersistentMessageReceiver
+
+from opentelemetry import propagate
+from opentelemetry import context
+from opentelemetry import trace, baggage
+from opentelemetry.semconv.trace import SpanAttributes, MessagingDestinationKindValues
+from opentelemetry.trace import StatusCode, Status, SpanKind
+from typing import List
+
+from solace_otel.messaging.trace.propagation import InboundMessageCarrier, InboundMessageGetter, OutboundMessageCarrier, OutboundMessageSetter
+
+TRACER_NAME = "my_tracer"
+OTEL_MESSAGE = "This is an OTel message"
+BAGGAGE_KEY_1 = "key1"
+BAGGAGE_VALUE_1 = "value1"
+BAGGAGE_KEY_2 = "key2"
+BAGGAGE_VALUE_2 = "value2"
+
+def function_that_might_raise_exception(raise_exception: bool=False):
+    """
+    This function simulates application-specific message processing which might raise an exception.
+    """
+    if raise_exception:
+        raise Exception("Something went wrong while processing the message.")
+
+class OTelMessageHandler(MessageHandler):
+    def __init__(self, receiver=None, ack=True):
+        super().__init__()
+        self._receive_count = 0
+        self._exception_list = []
+        self._receiver = receiver
+        self._ack = ack
+        self._baggage = None
+
+    def on_message(self, message: 'InboundMessage'):
+        try:
+            tracer = trace.get_tracer(TRACER_NAME)
+            PROPAGATOR = propagate.get_global_textmap()
+             
+            carrier = InboundMessageCarrier(message)
+
+            # inject context form a into the message using a carrier, context is implicit
```

Comment:
is this comment supposed to say "inject contest from the incoming message into the message"